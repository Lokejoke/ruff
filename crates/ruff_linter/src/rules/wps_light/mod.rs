//! Rules from [wps-light](https://pypi.org/project/wps-light/).
pub(crate) mod rules;

#[cfg(test)]
mod tests {
    use std::convert::AsRef;
    use std::path::Path;

    use anyhow::Result;
    use test_case::test_case;

    use crate::registry::Rule;
    use crate::test::test_path;
    use crate::{assert_messages, settings};

    #[test_case(Rule::ListMultiplication, Path::new("WPS435.py"))]
    #[test_case(Rule::ReservedArgumentAsVariable, Path::new("WPS117.py"))]
    #[test_case(Rule::UnderscoresInNumber, Path::new("WPS303.py"))]
    #[test_case(Rule::AssignmentToSubscriptSlice, Path::new("WPS362.py"))]
    fn rules(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.as_ref(), path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("wps_light").join(path).as_path(),
            &settings::LinterSettings::for_rule(rule_code),
        )?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }
}
