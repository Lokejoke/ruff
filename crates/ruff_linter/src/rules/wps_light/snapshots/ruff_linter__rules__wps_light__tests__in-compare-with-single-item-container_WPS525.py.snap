---
source: crates/ruff_linter/src/rules/wps_light/mod.rs
snapshot_kind: text
---
WPS525.py:2:7: WPS525 Avoid comparing to a single item container with membership operator
  |
2 | x1 in {"TEST"}  # [WPS525]
  |       ^^^^^^^^ WPS525
3 | 
4 | x2 in ["TEST"]  # [WPS525]
  |

WPS525.py:4:7: WPS525 Avoid comparing to a single item container with membership operator
  |
2 | x1 in {"TEST"}  # [WPS525]
3 | 
4 | x2 in ["TEST"]  # [WPS525]
  |       ^^^^^^^^ WPS525
5 | 
6 | x3 in {"x": "TEST"}  # [WPS525]
  |

WPS525.py:6:7: WPS525 Avoid comparing to a single item container with membership operator
  |
4 | x2 in ["TEST"]  # [WPS525]
5 | 
6 | x3 in {"x": "TEST"}  # [WPS525]
  |       ^^^^^^^^^^^^^ WPS525
7 | 
8 | single = ["TEST"]
  |

WPS525.py:17:7: WPS525 Avoid comparing to a single item container with membership operator
   |
15 | x5 in single
16 | 
17 | x6 in [single]  # [WPS525]
   |       ^^^^^^^^ WPS525
18 | 
19 | x7 in [*single]
   |

WPS525.py:21:27: WPS525 Avoid comparing to a single item container with membership operator
   |
19 | x7 in [*single]
20 | 
21 | x8 in ["TEST", "TEST"] in ["TEST"]  # [WPS525]
   |                           ^^^^^^^^ WPS525
22 | 
23 | x9 in ["TEST"] in ["TEST", "TEST"]  # [WPS525]
   |

WPS525.py:23:7: WPS525 Avoid comparing to a single item container with membership operator
   |
21 | x8 in ["TEST", "TEST"] in ["TEST"]  # [WPS525]
22 | 
23 | x9 in ["TEST"] in ["TEST", "TEST"]  # [WPS525]
   |       ^^^^^^^^ WPS525
24 | 
25 | 2 < x10 in [3]  # [WPS525]
   |

WPS525.py:25:12: WPS525 Avoid comparing to a single item container with membership operator
   |
23 | x9 in ["TEST"] in ["TEST", "TEST"]  # [WPS525]
24 | 
25 | 2 < x10 in [3]  # [WPS525]
   |            ^^^ WPS525
26 | 
27 | x10 in (2,)  # [WPS525]
   |

WPS525.py:27:8: WPS525 Avoid comparing to a single item container with membership operator
   |
25 | 2 < x10 in [3]  # [WPS525]
26 | 
27 | x10 in (2,)  # [WPS525]
   |        ^^^^ WPS525
28 | 
29 | x11 in {"1": 1, "2": 2}
   |

WPS525.py:39:8: WPS525 Avoid comparing to a single item container with membership operator
   |
37 | x15 in [x for x in range(1)]
38 | 
39 | x16 in [1] in [2]  # 2x [WPS525]
   |        ^^^ WPS525
40 | 
41 | 17 in [17] in [[17]]  # 2x [WPS525]
   |

WPS525.py:39:15: WPS525 Avoid comparing to a single item container with membership operator
   |
37 | x15 in [x for x in range(1)]
38 | 
39 | x16 in [1] in [2]  # 2x [WPS525]
   |               ^^^ WPS525
40 | 
41 | 17 in [17] in [[17]]  # 2x [WPS525]
   |

WPS525.py:41:7: WPS525 Avoid comparing to a single item container with membership operator
   |
39 | x16 in [1] in [2]  # 2x [WPS525]
40 | 
41 | 17 in [17] in [[17]]  # 2x [WPS525]
   |       ^^^^ WPS525
   |

WPS525.py:41:15: WPS525 Avoid comparing to a single item container with membership operator
   |
39 | x16 in [1] in [2]  # 2x [WPS525]
40 | 
41 | 17 in [17] in [[17]]  # 2x [WPS525]
   |               ^^^^^^ WPS525
   |
